src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -184,6 +184,13 @@ public EventTypeService(final EventTypeRepository eventTypeRepository,
             final EventType eventType = schemaEvolutionService.evolve(original, eventTypeBase);
             eventType.setDefaultStatistic(
                     validateStatisticsUpdate(original.getDefaultStatistic(), eventType.getDefaultStatistic()));
+            final Long newRetentionTime = eventTypeBase.getOptions().getRetentionTime();
+            final Long oldRetentionTime = original.getOptions().getRetentionTime();
+            if (newRetentionTime != null && !newRetentionTime.equals(oldRetentionTime)) {
```

Comment:
Here, it is possible that the change in retention time will succeed, but the event type will fail to update later (line 194). In this case the client will think everything has failed, even though the retention time was updated.

I think it might be better to change the retention time after updating the event type.